test/unit/lib/TopStoriesFeed.test.js

```diff
@@ -1460,4 +1460,41 @@ describe("Top Stories Feed", () => {
     assert.calledOnce(instance.uninit);
     assert.calledOnce(instance.init);
   });
+  describe("#layout", () => {
+    it("should call dispatchLayoutUpdate from fetchStories", async () => {
+      instance.stories_endpoint = "stories-endpoint";
+      let fetchStub = globals.sandbox.stub();
+
+      const response = {
+        "layout": [1, 2],
+      };
+      globals.set("fetch", fetchStub);
+      fetchStub.resolves({ok: true, status: 200, json: () => Promise.resolve(response)});
+      sinon.spy(instance, "dispatchLayoutUpdate");
+
+      await instance.fetchStories();
+      assert.calledOnce(instance.dispatchLayoutUpdate);
+      assert.calledWith(instance.dispatchLayoutUpdate, [1, 2]);
+    });
+    it("should call dispatchLayoutUpdate from loadCachedData", async () => {
+      sinon.spy(instance, "dispatchLayoutUpdate");
+      instance.cache.get = () => ({stories: {layout: [2, 3]}});
+
+      await instance.loadCachedData();
+      assert.calledOnce(instance.dispatchLayoutUpdate);
+      assert.calledWith(instance.dispatchLayoutUpdate, [2, 3]);
+    });
+    it("should call dispatch from dispatchLayoutUpdate with available data", () => {
+      instance.dispatchLayoutUpdate([1, 2]);
+      assert.calledOnce(instance.store.dispatch);
+      const [action] = instance.store.dispatch.firstCall.args;
+      assert.equal(action.type, "CONTENT_LAYOUT");
+      assert.equal(action.data[0], 1);
+      assert.equal(action.data[1], 2);
```

Comment:
These last two lines would be more readable if written as:

```
assert.equal(action.data, [1, 2])
```

Reply:
This didn't actually work, I think I need some sort of deep equal? I'll try that.

Reply:
Ah, gotta love complex JS equality semantics.  :-)